cloud_provider/aws/aws_minion_manager.py

```diff
@@ -635,7 +656,28 @@ def check_scaling_group_instances(self, scaling_group):
                 # Wait for sometime before checking again.
                 time.sleep(60)
         return False
-
+    
+    def check_insufficient_capacity(self, scaling_group):
+        """
+        Checks whether not completed ASG activities got not have sufficient capacity error message.
+        """
+        # This error message from https://docs.aws.amazon.com/autoscaling/ec2/userguide/ts-as-capacity.html#ts-as-capacity-1
+        INSUFFICIENT_CAPACITY_MESSAGE = ['We currently do not have sufficient',
+                                           'capacity in the Availability Zone you requested']
+        
+        asg_info = scaling_group.get_asg_info()
+        response = AWSMinionManager.describe_asg_with_retries(
+            self._ac_client, asg_info.AutoScalingGroupName)
+        activities = response.Activities
+        
+        for activity in activities:
+            if activity.Progress == 100:
+                continue
+            if len([message for message in INSUFFICIENT_CAPACITY_MESSAGE if message in activity.StatusMessage]) == len(INSUFFICIENT_CAPACITY_MESSAGE):
```

Comment:
Not every `activity` object will have `StatusMessage`. Can you check first if `StatusMessage` exists?

Reply:
Sorry, I don't know have case without `StatusMessage`. Added check and test case for it.